### ISSUE TITLE
Harpoon prints unparsible deBrujin index of a substitution when using the suffices command implicitly

### ISSUE THREAD

**Issue by Ailrun** [Beluga-lang/Beluga#210]
The last theorem in `t/harpoon/poplmark-reloaded/props.input`, i.e.
https://github.com/Beluga-lang/Beluga/blob/b93d57a4fcc4675a3a9d9f6721353193f622576e/t/harpoon/poplmark-reloaded/props.input#L74-L80
reveals the problem.
When executing this, Harpoon prints
```
proof Sn-appl : (g : cxt)
                  {M : (g |- tm (arr A[] B[]))}
                    {N : (g |- tm A[])} Sn [g |- app M N] -> Sn [g |- M] =
/ total /
intros
{ g : cxt,
  A : ( |- ty),
  B : ( |- ty),
  M : (g |- tm (arr A[] B[])),
  N : (g |- tm A[])
| x : Sn [_ |- app M N]
; split x as
  case Acc:
  { g : cxt,
    A : ( |- ty),
    B : ( |- ty),
    M : (g |- tm (arr A[] B[])),
    N : (g |- tm A[])
  | x : Sn [_ |- app M N],
    x1 : {M' : (g |- tm B[])} {S : (g |- step (app M N) M')} Sn [_ |- M']
  ; suffices by Acc [_] [ |- arr A B[^0][]] [_ |- M[^0][..]] toshow
% ...
```
and gives an error saying
```
t/harpoon/poplmark-reloaded/tps.bel, <position of 0 inside of arr A B[^0]>:
Parse error.
Unexpected token in stream
  Expected token `]'
  Got token `0'
```

**Comment by tsani** [comment]
These happen because the term/type is not normalized. In reality, the `M[^0][..]` is an instantiated MMVar and the `[^0]` is its associated msub. Normalization (`cnorm*`) will eliminate these. I encountered this problem myself when using suffices and should I had fixed it... Anyway, the key thing is that when constructing the proof script syntax for the `suffices`, we need to ensure that all the type annotations are normalized.

I just pushed ffc51b18 which adds ensures that types elaborated via `Interactive.elaborate_typ` (e.g. those in the `suffices` command) are normalized. Let me know if this fixes the issue.

**Comment by tsani** [comment]
Actually upon closer inspection that's not the issue. The problem is related to the generation of the MApps to eliminate the leading PiBoxes. They need to be normalized to get rid of instantiated MMVars.

It should be enough to normalize the suffices scrutinee *after* unifying with the type annotations, before checking for closedness. Gimme a minute.

**Comment by tsani** [comment]
Okay, this should be it.

I added normalization of the scrutinee of the suffices in 200a0b2b. Try this and let me know.

**Comment by Ailrun** [comment]
It still generates M[^0][..] in other context, namely,
```
norm
{g : cxt} {M : [g |- tm A[]]} SN [g |- M]


suffices by cr1 toshow _
suffices by fundamental_lemma toshow RedS [g] [g |- ..]
solve id_red [g]
```
with `t/harpoon/poplmark-reloaded/tps.bel`.

**Comment by Ailrun** [comment]
Since this is avoidable (by using `solve` instead of `suffices`), I moved this to v2

**Comment by Ailrun** [comment]
This is caused by some unsolved constraint. Maybe we need to check whether all constraints are solved after each of the commands.

**Comment by Ailrun** [comment]
Now `suffices` command checks existence of leftover constraints. So in the case of this issue, it will throw an error on the command before a user actually finishes a proof.

**Comment by Ailrun** [comment]
Fixed by 59a02f7
